src/screens/Sablier/Withdraw.js

```diff
@@ -0,0 +1,222 @@
+  const assetData = getAssetDataByAddress(getAssetsAsList(assets), supportedAssets, token.id);
+
+  useEffect(() => {
+    calculateSablierWithdrawTransactionEstimate(stream, withdrawAmountInWei, assetData);
+  }, [withdrawAmountInWei]);
+
+  useEffect(() => {
+    // the exact max amount to withdraw must be fetched from the blockchain
+    // as we can only estimate due to the fact that the contract uses block.timestamp
+    fetchStreamBalance(streamId, recipient)
+      .then((streamBalance) => {
+        setMaxWithdraw(streamBalance);
+        setIsFetchingMaxWithdraw(false);
+      })
+      .catch(() => {});
+  }, []);
+
+  const goToReviewScreen = () => {
+    navigation.navigate(SABLIER_WITHDRAW_REVIEW, {
+      withdrawAmountInWei,
+      stream,
+    });
+  };
+
+  const onValueChanged = (value: Object) => {
+    if (!value) {
+      setWithdrawAmount(0);
+      return;
+    }
+    const valueInWei = utils.parseUnits(value?.input, assetData.decimals);
+    if (valueInWei.gt(maxWithdraw)) {
+      setWithdrawAmount(maxWithdraw);
+    } else {
+      setWithdrawAmount(valueInWei);
+    }
+  };
+
+  const assetOptions = [assetData];
+
+  const streamedAssetBalance: Balances = {
+    [assetData.symbol]: { symbol: assetData.symbol, balance: formatUnits(maxWithdraw, assetData.decimals) },
+  };
+
+  const txFeeInfo = buildTxFeeInfo(withdrawTransactionEstimate, useGasToken);
+  const isEnoughForFee = !!txFeeInfo?.fee && isEnoughBalanceForTransactionFee(balances, {
+    txFeeInWei: txFeeInfo.fee,
+    gasToken: txFeeInfo.gasToken,
+  });
+
+  const isNextButtonDisabled = isCalculatingWithdrawTransactionEstimate
+    || isFetchingMaxWithdraw
+    || !withdrawAmountInWei
+    || !isEnoughForFee
+    || (!!txFeeInfo?.fee && !txFeeInfo.fee.gt(0));
+  const nextButtonTitle = isCalculatingWithdrawTransactionEstimate ? 'Getting fee..' : 'Next';
+
+  return (
+    <ContainerWithHeader
+      headerProps={{ centerItems: [{ title: 'Withdraw' }] }}
+      footer={
+        <FooterWrapper>
+          <FeeLabelToggle
+            labelText="Fee"
+            txFeeInWei={txFeeInfo?.fee}
+            gasToken={txFeeInfo?.gasToken}
+            isLoading={isCalculatingWithdrawTransactionEstimate}
+            showFiatDefault
+          />
+          <Spacing h={16} />
+          <Button
+            disabled={isNextButtonDisabled}
+            title={nextButtonTitle}
+            onPress={goToReviewScreen}
+            block
+          />
+        </FooterWrapper>
+      }
+    >
+      <ValueSelectorCard
+        preselectedAsset={assetData.symbol}
+        customOptions={assetOptions}
+        balances={streamedAssetBalance}
+        baseFiatCurrency={baseFiatCurrency}
+        rates={rates}
+        maxLabel="Max"
+        getFormValue={onValueChanged}
+        isLoading={isFetchingMaxWithdraw}
+      />
+      <BaseText regular secondary center>You will receive {assetData.symbol} in your wallet.</BaseText>
```

Comment:
Just a note - we're moving away from the language of referring to the app as a "wallet". Nothing to action here, but i just wanted to let you know.